PRINCIPLES.md

```diff
@@ -1,74 +1,22 @@
-
-- ### Desired State
-
-    The aggregate of all configuration data for a system form its _Desired State_ which is defined as data sufficient to recreate the system so that instances of the system are behaviourally indistinguishable.
+The [desired state](./GLOSSARY.md#desired-state) of a GitOps managed system must be:
 
-- ### Drift
+1. **Declarative**
 
-    When a system's _Actual State_ changes for any reason other than its versioned _Desired State_ declarations having changed, we say that the system has drifted from its _Desired State_.
+    A [system](./GLOSSARY.md#software-system) managed by GitOps must have its desired state expressed [declaratively](./GLOSSARY.md#declarative-description).
 
-- ### Reconciliation
+2. **Versioned and Immutable**
 
-    The process of ensuring that the _Actual State_ of a sytem matches its versioned _Desired State_ declarations.
-    Contrary to CIops, any divergence between the two will trigger reconciliation, regardless of where changes occured.
-    Divergence could be due to the actual state unintentionally _Drifting_ from the desired state declarations, or a new desired state declaration version having been changed intentionally.
+    Desired state is [stored](./GLOSSARY.md#state-store) in a way that enforces immutability, versioning and retains a complete version history.
 
-- ### Software System
+3. **Pulled Automatically**
 
-    One or more Runtime environments consisting of resources under management.
-    In each Runtime, management Agents to act on resources according to security policies.
-    One or more software Repositories for storing deployable artifacts that may be loaded into the runtime environments, eg. configuration files, code, binaries and packages.
-    One or more Administrators who are responsible for operating the runtime environments ie. installing, starting, stopping and updating software, code, configuration, etc.
-    A set of policies controlling access and management of repositories, deployments, runtimes.
+    Software agents automatically pull the desired state declarations from the source.
 
-- ### State Store
+4. **Continuously Reconciled**
 
-    A system for storing immutable versions of _Desired State_ declarations.
-    This state store should provide access control and auditing on the changes to the Desired State.
-    Git is the canonical example used as this State Store, and where GitOps derived its name, but but any other system that meets this criteria may be used.
-    In all cases these must be properly configured, and special precautions must be taken to comply with requirements set out in the GitOps Principles.
+    Software agents [continuously](./GLOSSARY.md#continuous) observe actual system state and [attempt to apply](./GLOSSARY.md#reconciliation) the desired state.
```

Comment:
It's not really a case of _applying_ the desired state. Typically, the desired state is abstract (e.g., a Kubernetes Deployment) and the observed state is concrete (these pods are running). Reconciliation takes actions to move the concrete state so that it fulfils the abstract model.

You might ask "but isn't that something Kubernetes does, rather than the GitOps system?" Kubernetes is part of the GitOps system -- you need something (e.g., Flux) to convey the abstract, desired state to Kubernetes, _and_ Kubernetes to interpret it into a running system. Without Kubernetes doing its job, you're just pasting files into a database; and without e.g., Flux doing its job, you don't get the benefit of working via Git.

Reply:
@squaremo I think we've closed changes to the principles themselves in #21 unless it's a major issue. 

For this particular case, we used the language "attempt to apply" because it's both broader and doesn't require success. For example, if the desired state requires more nodes than the physical system has available then an attempt to apply could simply mean the software agent makes the request of Kubernetes but could include emailing and alerting an admin that they're out of nodes. 

Of course, if we weren't using Kubernetes, then attempt to apply could mean a whole different set of procedures in order to attempt to apply the desired state.

Reply:
> @squaremo I think we've closed changes to the principles themselves in #21 unless it's a major issue.

@todaywasawesome quick note on this, I don't believe we agreed to close changes to the principles. But instead agreed to a RC/feedback approach, to give folks a chance to give feedback on RC 1, leading up to the [planned RC 2 this Friday](https://github.com/open-gitops/documents/milestone/7), and planned [full release by Oct 11](https://github.com/open-gitops/documents/milestone/1). Right?

@squaremo do you have a concrete suggestion for another way to phrase this other than attempting to "apply", which does not only refer to Kubernetes?

Reply:
> do you have a concrete suggestion for another non-kubernetes-specific way to phrase this other than attempting to "apply" the desired state to the actual state?

"Taking action to make progress towards the desired state"

Reply:
OK, for now changing "attempt to apply" to "attempt to reach the desired state" (simpler wording. Alternatively "attempt to make progress toward the desired state" is longer but also fine). Doing this now so that we can stay on target for publishing RC 2 today.

Because this is important, @todaywasawesome will open a discussion just on this point today, CC @squaremo, and link to this comment thread for context https://github.com/open-gitops/documents/pull/22#discussion_r717644615.

I believe the full "Taking action to make progress towards the desired state" is relevant to the possible 5th principle that was temporarily excluded from the RC:

>5. Operated in a closed loop
>Software agents observe desired state and meta-state to take actions based on policy when the desired state cannot be reached. This may include things like notifications, rollbacks, etc.

We did not include this because it's not clear whether a closed (feedback) loop is _required_ for GitOps. If not, it should not be a foundational principle. If so, it should. If we're on the fence about that as a group, this could all perhaps be addressed in the glossary.

Can we follow-up in that forthcoming discussion? If we can resolve that question in time, it could make it into the first full release (v1.0.0) scheduled to be ready by KubeCon NA. If not, it could be added to future releases.

Reply:
To be continued in https://github.com/open-gitops/documents/issues/24 😸